packages/synapse-react-client/src/components/ChallengeDetailPage/ChallengeDetailPage.tsx

```diff
@@ -47,7 +49,14 @@ export function ChallengeDetailPage({ projectId }: ChallengeDetailPageProps) {
         challenge.participantTeamId,
         userProfile.ownerId,
         accessToken,
-      )
+      ).then(() => {
+        queryClient.invalidateQueries(
+          keyFactory.getIsUserMemberOfTeamQueryKey(
+            challenge.participantTeamId,
+            userProfile.ownerId,
+          ),
+        )
+      })
```

Comment:
If you have time, would prefer to refactor so that we have a `useDeleteTeamMembership` mutation colocated  with the other react-query hooks, which internally handles invalidation in the `onSuccess` callback.

Then the hooks are more reusable and we don't need to worry about invalidation at the component level. See `useDeleteThread` in useThread.tsx for an example